crates/chain/src/keychain/txout_index.rs

```diff
@@ -559,32 +559,75 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     }
 }
 
-fn range_descriptor_spks<'a, R>(
+fn range_descriptor_spks(
+    descriptor: Cow<'_, Descriptor<DescriptorPublicKey>>,
+    range: Range<u32>,
```

Comment:
Instead of taking in a concrete type `Range` (I know I recommended this, I'm sorry), take in a generic `RangeBounds`.

https://doc.rust-lang.org/core/ops/trait.RangeBounds.html

This way you can still use `a..=b`